src/script/links/LinkPreviewRepository.js

```diff
@@ -143,6 +143,20 @@ z.links.LinkPreviewRepository = class LinkPreviewRepository {
    * @returns {Promise} Resolves with the retrieved open graph data
    */
   _fetchOpenGraphData(link) {
+    if (typeof window.openGraphAsync === 'function') {
+      return window
+        .openGraphAsync(link)
+        .then(data => {
+          if (data) {
```

Comment:
The logic seems to be exactly the same for both handler. Can we extract it into a new function